### PR TITLE
Make ConvertTo-Html work on CoreCLR

### PR DESCRIPTION
**Pull request by jeffbi**
Part of issue #2240 

This also includes some Pester test for the cmdlet.

### CODE REVIEW

File: test/powershell/Modules/Microsoft.PowerShell.Utility/ConvertTo-Html.Tests.ps1
Comment (vors) [inline test/powershell/Modules/Microsoft.PowerShell.Utility/ConvertTo-Html.Tests.ps1]: We are trying to move away from the Windows BVT/DRT terminology.
Please, just use the descriptive names for the `Describe`. More testing guidelines are here: https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/WritingPesterTests.md#writing-pester-tests

File: src/Microsoft.PowerShell.Commands.Utility/project.json
Comment (vors) [inline src/Microsoft.PowerShell.Commands.Utility/project.json]: I wonder, why was it taking away at the first place. @daxian-dbw @andschwa ?

**Comment by vors** [comment]
Hi @jeffbi ! Thank you for your contribution.

Based on CI logs from appveyor, looks like you hit the problem that was discussed here https://github.com/PowerShell/PowerShell/pull/2182#discussion_r78429825

Multi-line strings are not very reliable for testing, because they depend on OS and the way how test file itself was checkout (for instance on AppVeyor all line ends for git files are `\n`, not `\r\n`).

In general, it's better to avoid using multi-line lines in the tests.
This case seems like an exception, where it's reasonable.
You can add code to strip-out line endings, similar to this:

https://github.com/PowerShell/platyPS/blob/32cc4b625348e40ce39dd12da352eabc203720f9/test/Pester/PlatyPs.Tests.ps1#L612

```
[-] Test ConvertTo-Html with no parameters 100ms
   Expected string length 386 but was 396. Strings differ at index 110.
   Expected: {<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN"  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">\n<html xmlns="http://www.w3.org/1999/xhtml">\n<head>\n<title>HTML TABLE</title>\n</head><body>\n<table>\n<colgroup><col/><col/><col/></colgroup>\n<tr><th>Name</th><th>Age</th><th>Friends</th></tr>\n<tr><td>John Doe</td><td>42</td><td>System.Object[]</td></tr>\n</table>\n</body></html>}
   But was:  {<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN"  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">\r\n<html xmlns="http://www.w3.org/1999/xhtml">\r\n<head>\r\n<title>HTML TABLE</title>\r\n</head><body>\r\n<table>\r\n<colgroup><col/><col/><col/></colgroup>\r\n<tr><th>Name</th><th>Age</th><th>Friends</th></tr>\r\n<tr><td>John Doe</td><td>42</td><td>System.Object[]</td></tr>\r\n</table>\r\n</body></html>}
```

File: src/Microsoft.PowerShell.Commands.Utility/project.json
Comment (daxian-dbw) [inline src/Microsoft.PowerShell.Commands.Utility/project.json]: It was never taken in. I guess we just didn't get to it.

File: test/powershell/Modules/Microsoft.PowerShell.Utility/ConvertTo-Html.Tests.ps1
Comment (mirichmo) [inline test/powershell/Modules/Microsoft.PowerShell.Utility/ConvertTo-Html.Tests.ps1]: This test code should go within BeforeAll instead of existing within the Describe. See [Pester Do's and Don'ts](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/PesterDoAndDont.md)

Comment (jeffbi) [inline test/powershell/Modules/Microsoft.PowerShell.Utility/ConvertTo-Html.Tests.ps1]: Done. New commit pushed up.

**Comment by mirichmo** [comment]
LGTM.

I'll merge it once we get resolution on the CLA issue.

**Comment by jeffbi** [comment]
The CLA has been signed. Is there still an issue with it?

**Review by mirichmo — CHANGES_REQUESTED** [review]
Blocking this PR pending resolution of the CLA issue

**Review by mirichmo — APPROVED** [review]

**Pull request merged**
